src/components/ProjectsPage/Projects.js

```diff
@@ -40,12 +42,46 @@ const Projects = ({
   const [convertedYaml, setConvertedYaml] = useState('')
   const [createProject, setCreateProject] = useState(false)
   const [filteredProjects, setFilteredProjects] = useState([])
+  const [filterByName, setFilterByName] = useState('')
+  const [filterMatches, setFilterMatches] = useState([])
+  const [isDescendingOrder, setIsDescendingOrder] = useState(true)
   const [isEmptyValue, setIsEmptyValue] = useState(false)
   const [selectedProjectsState, setSelectedProjectsState] = useState(
     'allProjects'
   )
+  const [sortProjectId, setSortProjectId] = useState('byName')
 
-  const projectsStates = useMemo(generateProjectsStates, [])
+  const isValidProjectState = useCallback(
+    project => {
+      return (
+        (selectedProjectsState === 'allProjects' &&
+          project.status.state !== 'archived') ||
+        project.status.state === selectedProjectsState
+      )
+    },
+    [selectedProjectsState]
+  )
+
+  const handleFilterProject = useCallback(
+    project => {
+      return filterByName.length > 0
+        ? project.metadata.name.includes(filterByName) &&
+            isValidProjectState(project)
+        : isValidProjectState(project)
+    },
+    [filterByName, isValidProjectState]
+  )
+
+  const handleSortProjects = useCallback(
+    projects => {
+      const sortPath = projectsSortOptions.find(
+        option => option.id === sortProjectId
+      ).path
+
+      return orderBy(projects, [sortPath], [isDescendingOrder ? 'asc' : 'desc'])
```

Comment:
This seems to be the opposite way we expect. When the arrow points down we want it to have descending order,  when the arrow is up we want it to be ascending order.

```suggestion
      return orderBy(projects, [sortPath], [isDescendingOrder ? 'desc' : 'asc'])
```